src/utils/dateUtils/historicalStockData.js

```diff
@@ -0,0 +1,31 @@
+import {
+  MONTHS_PER_YEAR,
+  VALID_DATE_NOT_FOUND
+} from '../../constants';
+
+import {
+  calculateDateDaysInPast,
+  calculateDateMonthsInPast,
+  formatDateForMaxStockData
+} from '.';
+
+export const getStockDataForPreviousMonths = (maxStockData, months) => {
+  const date = new Date();
+  let unformattedCutoffDate = calculateDateMonthsInPast(date, months);
+  let cutoffDate;
+  let elementPosition = VALID_DATE_NOT_FOUND;
+
+  // If the date we are looking for is a weekend (and thus the stock has no data),
+  // keep checking one day before that until a day with stock data is found
+  while (elementPosition === VALID_DATE_NOT_FOUND) {
+    cutoffDate = formatDateForMaxStockData(unformattedCutoffDate);
+    elementPosition = maxStockData.map(data => {
```

Comment:
I think this could shorten to:
`elementPosition = maxStockData.map(data => data.date).indexOf(cutoffDate);`